community/detectors/apache_airflow_cve_2020_17526/src/main/java/com/google/tsunami/plugins/cve202017526/Cve202017526Detector.java

```diff
@@ -0,0 +1,234 @@
+            if (cookie.getName().equals("session")) {
+                freshSessionCookieValue = cookie.getValue();
+            }
+        }
+        if (freshSessionCookieValue == null) {
+            return null;
+        }
+        results.put("freshSessionCookieValue", freshSessionCookieValue);
+
+        Pattern CSRF_PATTERN = Pattern.compile("var CSRF = \"([\\d\\w-.]+)\"");
+        Matcher m = CSRF_PATTERN.matcher(firstResponse.bodyString().get());
+        if (!m.find()) {
+            return null;
+        }
+        String freshCsrfToken = m.group(1);
+        results.put("freshCsrfToken", freshCsrfToken);
+        logger.atWarning().log(results.get("freshCsrfToken"));
+        logger.atWarning().log(results.get("freshSessionCookieValue"));
+        return results;
+    }
+
+    private Payload getTsunamiCallbackHttpPayload() {
+        try {
+            return this.payloadGenerator.generate(
+                    PayloadGeneratorConfig.newBuilder()
+                            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.BLIND_RCE)
+                            .setInterpretationEnvironment(
+                                    PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+                            .setExecutionEnvironment(
+                                    PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+                            .build());
+        } catch (NotImplementedException n) {
+            return null;
+        }
+
+    }
+
+    private DetectionReport buildDetectionReport(
+            TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+        return DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(vulnerableNetworkService)
+                .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                        Vulnerability.newBuilder()
+                                .setMainId(
+                                        VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                                                .setValue("CVE-2020-17526"))
+                                .setSeverity(Severity.CRITICAL)
+                                .setTitle("CVE-2020-17526 Authentication bypass lead to Arbitrary Code Execution in Apache Airflow")
+                                .setDescription(
+                                        "An attacker can use a default DAG to execute arbitrary code on"
+                                                + " the server hosting the apache airflow application.")
+                                .setRecommendation(
+                                        "update to higher versions or Change default value for [webserver] secret_key config"))
```

Comment:
Please consider providing recommendations regarding an upgrade to a specific non-vulnerable version of Apache Airflow